test/visitor/loop_unroll.cpp

```diff
@@ -34,6 +35,10 @@ std::string run_loop_unroll_visitor(const std::string& text) {
     ConstantFolderVisitor().visit_program(ast.get());
     LoopUnrollVisitor().visit_program(ast.get());
     ConstantFolderVisitor().visit_program(ast.get());
+
+    // check that, after visitor rearrangement, parents are still up-to-date
+    CkParentVisitor(true).visit_program(ast.get());
```

Comment:
I would like to bit discuss about this `CkParentVisitor` visitor and how this validates if the tree being visited is always consistent.

Reply:
yes, ofc